internal/proxy/env.go

```diff
@@ -36,37 +35,30 @@ func (p Proxy) applyToEnvironment() (err error) {
 	}
 
 	log.Debugf("Applying environment proxy configuration to %q", p.envConfigPath)
-	content := fmt.Sprintln(confHeader)
-	for _, p := range p.settings {
-		content += p.envString()
-	}
 
-	if prevContent, err := previousConfig(p.envConfigPath); err == nil && prevContent == content {
+	contents := p.envFileContents()
+	if prevContent, err := previousConfig(p.envConfigPath); err == nil && prevContent == contents {
 		log.Debugf("Environment proxy configuration at %q is already up to date", p.envConfigPath)
 		return nil
 	} else if err != nil && !errors.Is(err, fs.ErrNotExist) {
 		return err
 	}
 
 	// Check if the parent directory exists - attempt to create the structure if not
-	envConfigDir := filepath.Dir(p.envConfigPath)
-	if _, err := os.Stat(envConfigDir); errors.Is(err, fs.ErrNotExist) {
-		log.Debugf("Creating directory %q", envConfigDir)
-		// #nosec G301 - /etc/environment.d permissions are 0755, so we should keep the same pattern
-		if err := os.MkdirAll(envConfigDir, 0755); err != nil {
-			return fmt.Errorf("failed to create environment config parent directory: %w", err)
-		}
-	} else if err != nil {
-		return fmt.Errorf("unexpected error while checking environment directory: %w", err)
-	}
-
-	// #nosec G306 - /etc/environment.d/* permissions are 0644, so we should keep the same pattern
-	if err := os.WriteFile(p.envConfigPath+".new", []byte(content), 0644); err != nil {
+	if err := createParentDirectory(p.envConfigPath); err != nil {
 		return err
 	}
-	if err := os.Rename(p.envConfigPath+".new", p.envConfigPath); err != nil {
-		return err
+
+	return writeConfig(p.envConfigPath, contents)
+}
+
+// envFileContents returns the content of the environment proxy configuration
+// file to be written.
+func (p Proxy) envFileContents() string {
+	contents := fmt.Sprintln(confHeader)
+	for _, p := range p.settings {
+		contents += p.envString()
```

Comment:
I could suggest using a function as parameter, but then, it would be too functional for my taste :p